### PR TITLE
NXP-28885: Retention Search feature is not working

### PR DESCRIPTION
**Pull request by RSalem07**
PR opened manually, at the same time the T&P is launched (https://qa2.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-saouana-10.10/111/) -> **T&P OK**

As discussed with @jaubenque,  PO needs to release a first version of retention addon asap. 

Its why: 
Currently there is no Tests coverture on retention search feature at the backend part (test of the Page Provider). As see with @jaubenque he will create for us a new ticket to make improvements on retention and we will add the needed tests.

For the web-ui a new ticket will be created to improve the way to make the search auto vs manually (the search button)  until that we make choice to handle it with the search button.

### CODE REVIEW

**Comment by efge** [comment]
Please update the commit message, it should not say what's not working but what's being done/fixed. For instance `NXP-28885: fix retention search`

**Comment by efge** [comment]
Also in the future please put `10.10` somewhere in the name of your branch if it's a 10.10 branch.

File: addons/nuxeo-retention/nuxeo-retention-web-ui/src/main/resources/web/nuxeo.war/ui/search/retention/nuxeo-retention-search-form.html [hunk 1]
```diff
@@ -50,7 +50,7 @@
                    value="{{searchTerm}}"
                    placeholder="[[i18n('defaultSearch.fullText.placeholder')]]"
                    autofocus
-                   on-keydown="_checkForEnter">
```
Comment (nmpcunha) [inline addons/nuxeo-retention/nuxeo-retention-web-ui/src/main/resources/web/nuxeo.war/ui/search/retention/nuxeo-retention-search-form.html:53]: Not sure we want to change this behaviour. Was it a requirement?

Comment (RSalem07) [reply addons/nuxeo-retention/nuxeo-retention-web-ui/src/main/resources/web/nuxeo.war/ui/search/retention/nuxeo-retention-search-form.html:53]: see my comment: https://github.com/nuxeo/nuxeo/pull/3907#issue-401422309.

Comment (nmpcunha) [reply addons/nuxeo-retention/nuxeo-retention-web-ui/src/main/resources/web/nuxeo.war/ui/search/retention/nuxeo-retention-search-form.html:53]: Thanks!
As discussed we can keep the `_checkForEnter` and add `auto` property to our `retentionSearch` contribution on `nuxeo-retention.html`

**Pull request merged**